### PR TITLE
documentation for the memory layout of the md-uv380

### PR DESCRIPTION
**Pull request by UnHolds**
added some documentation on how the linker file should look like.

### CODE REVIEW

File: mduv380.md [hunk 1]
```diff
@@ -15,6 +15,33 @@ __Model variants__
...
+ flash (rx) : ORIGIN = 0x0800C000, LENGTH = 1M - 48K - 128K /* 128k for settings */
+ ccm (rwx) : ORIGIN = 0x10000000, LENGTH = 64K
+}
+```
+
+When the stack pointer is defined it can only reside at the origin of the SRAM
+plus the length of the SRAM -4 or lower. This is because the stock bootloader
+has an off by one error when sanity checking the stack pointer.
+This means the stack can't be placed in the highest possible address in SRAM as
+is conventional, but rather it has to be at least one word down from the top.
```
Comment (Xiretza) [inline mduv380.md:38]: I had some difficulty understanding this paragraph - maybe something like this?


```suggestion
Usually, the stack pointer is defined to be one word past the addressable memory,
so the first `push` instruction writes to the very last memory location.

The stock bootloader performs a sanity check on the initial stack pointer, but
contains an off-by-one error: in order to boot, the initial stack pointer must
be *within* the SRAM region. Thus, the stack must be placed at least one word
down from the top.
```

**Review by Xiretza — COMMENTED** [review]
Thanks for documenting this!

File: mduv380.md (hunk 1 above)
Comment (UnHolds) [reply mduv380.md:38]: this sounds way better

File: mduv380.md [hunk 2]
```diff
@@ -15,6 +15,33 @@ __Model variants__
     * VHF: 136.000-174.000
     * UHF: 400.000-480.000
 
+## Memory Layout
+The STM32F405 of the MD-UV380 has 128k SRAM, 64k CCM, and 1M FLASH.
+The stock bootloader resides at the first 48k of the FLASH therefore we only
+have 1M - 48k (0xC000) bytes of FLASH. The last 128k bytes in the FLASH are
+reserverd for storing the settings.
```
[the remark this answers is not in the document]
Comment (UnHolds) [reply mduv380.md]: changed :)

File: mduv380.md [hunk 3]
```diff
@@ -15,6 +15,33 @@ __Model variants__
...
+
+Considering this the memory regions in the liker script looks like this:
```
[the remark this answers is not in the document]
Comment (UnHolds) [reply mduv380.md]: changed

**Comment by UnHolds** [comment]
fixed the typos and changed the explanation
thx for the comments @Xiretza

**Comment by silseva** [comment]
Just one note: the last 128kB flash sector is used only by OpenRTX as a temporary storage for the radio state at power off, so it should not be mentioned in the documentation. :)

**Comment by UnHolds** [comment]
A okay, I thought the stock firmware also puts some settings there. I will change it thx

**Pull request merged**